vessim/controller.py

```diff
@@ -9,23 +9,29 @@
 import pandas as pd
 
 from vessim.util import Clock
+from vessim.signal import Signal
 
 if TYPE_CHECKING:
     from vessim.cosim import Microgrid
 
 
 class Controller(ABC):
-    def __init__(self, step_size: Optional[int] = None):
+
+    def __init__(
+        self,
+        step_size: Optional[int] = None,
+        grid_signals: Optional[dict[str, Signal]] = None,
```

Comment:
I'd remove this here, not all controllers need access to grid_signals and if then it's cleaner if they manage the access by themselves